system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.cpp

```diff
@@ -28,31 +30,36 @@ namespace system_metrics_collector
 
 PeriodicMeasurementNode::PeriodicMeasurementNode(
   const std::string & name,
-  const std::chrono::milliseconds measurement_period,
-  const std::string & publishing_topic,
-  const std::chrono::milliseconds publish_period)
-: Node(name),
-  measurement_period_(measurement_period),
-  publishing_topic_(publishing_topic),
-  measurement_timer_(nullptr),
-  publish_timer_(nullptr),
-  publish_period_(publish_period)
+  const rclcpp::NodeOptions & options)
+: Node{name, options}
 {
-  // rclcpp::Node throws if name is empty
-
-  if (measurement_period <= std::chrono::milliseconds{0}) {
-    throw std::invalid_argument{"measurement_period cannot be negative"};
-  }
+  rcl_interfaces::msg::IntegerRange positive_range;
+  positive_range.from_value = 1;
+  positive_range.to_value =
+    std::numeric_limits<decltype(rcl_interfaces::msg::IntegerRange::to_value)>::max();
+  positive_range.step = 1;
 
-  if (publishing_topic.empty()) {
-    throw std::invalid_argument{"publishing_topic cannot be empty"};
-  }
+  rcl_interfaces::msg::ParameterDescriptor descriptor;
+  descriptor.read_only = true;
+  descriptor.integer_range.push_back(positive_range);
```

Comment:
does using `emplace_back` make sense here?

Reply:
I think the code would be:
```c++
descriptor.integer_range.emplace_back(
  /* from_value */ 1,
  /* to value */ std::numeric_limits<decltype(rcl_interfaces::msg::IntegerRange::to_value)>::max(),
  /* step value */ 1);
```
an alternative approach would be to use the `rvalue push_back` by doing:
```c++
descriptor.integer_range.push_back(std::move(positive_range));
```

IMO: readability is preferred over saving the cost of initializing+moving a struct

Reply:
I'll try, but if I recall correctly I typically find that these ROS `msg` objects typically can't be constructed that way.